src/zenml/stack_stores/sql_stack_store.py

```diff
@@ -268,27 +284,55 @@ def deregister_stack(self, name: str) -> None:
 
     # Private interface implementations:
 
-    def _create_stack(
-        self, name: str, stack_configuration: Dict[StackComponentType, str]
+    def _save_stack(
+        self,
+        name: str,
+        stack_configuration: Dict[StackComponentType, str],
+        is_update: bool = False,
     ) -> None:
-        """Add a stack to storage.
+        """Save a stack.
 
         Args:
             name: The name to save the stack as.
             stack_configuration: Dict[StackComponentType, str] to persist.
         """
         with Session(self.engine) as session:
-            stack = ZenStack(name=name, created_by=1)
-            session.add(stack)
-            for ctype, cname in stack_configuration.items():
-                if cname is not None:
-                    session.add(
-                        ZenStackDefinition(
-                            stack_name=name,
-                            component_type=ctype,
-                            component_name=cname,
+            if is_update:
+                # CHECK THAT THE STACK EXISTS FIRST
+                for ctype, cname in stack_configuration.items():
+                    if cname is not None:
+                        statement = (
+                            select(ZenStackDefinition)
+                            .where(ZenStackDefinition.stack_name == name)
+                            .where(ZenStackDefinition.component_type == ctype)
+                        )
+                        results = session.exec(statement)
+                        component = results.one()
```

Comment:
I believe `.one()` actually raises an exception if it didn't find anything and `.one_or_none()` is the one we should use here

Reply:
Seems like `.first()` would also work. `one_or_none()` will raise an error if there's ever more than one result returned. I think we would never have more than one result in this case.

Reply:
Ah yes `.first()` is probably equivalent, not sure if the additional check that only at least one exists has any value here, I'm fine with both.